src/test/java/definitions/EditWidgetDefinitions.java

```diff
@@ -0,0 +1,92 @@
+package definitions;
+
+import com.report.portal.business.constants.WidgetType;
+import com.report.portal.business.steps.DashboardsSteps;
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.*;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Map;
+
+import static com.report.portal.core.utils.data.RandomSequenceCreator.generateRandomString;
+
+@Getter
+@Setter
+public class EditWidgetDefinitions {
+
+    private DashboardsSteps dashboardsSteps = new DashboardsSteps();
+    private static final String RANDOM_WIDGET_NAME = generateRandomString(128);
+    private static final String RANDOM_WIDGET_DESC = generateRandomString(1500);
+
+    private WidgetType widgetType;
+    private String widgetName;
+    private String widgetDescription;
+
+    @Given("Widget is visible")
+    public void widgetIsVisible(DataTable dataTable) {
+        Map<String, String> widgetInfo = dataTable.asMap(String.class, String.class);
+        WidgetType widget = Enum.valueOf(WidgetType.class, widgetInfo.get("type"));
+        String widgetName = widgetInfo.get("name");
+        dashboardsSteps.assertWidgetVisible(widgetName, widget);
```

Comment:
assetions are usually going at the very end of the method
would it work with setters before assertion?